src/webapi/location.rs

```diff
@@ -1,38 +1,120 @@
 use webcore::value::Reference;
 use webcore::try_from::TryInto;
-use private::TODO;
+use webapi::dom_exception::SecurityError;
 
 /// The `Location` interface represents the location (URL) of the object it
 /// is linked to. Changes done on it are reflected on the object it relates
 /// to. Both the [Document](struct.Document.html) and [Window](struct.Window.html)
 /// interface have such a linked `Location`, accessible via [Document::location](struct.Document.html#method.location)
 /// and [Window::location](struct.Window.html#method.location) respectively.
 ///
+/// Note that all `Location` methods can return a `SecurityError` if the `Location` object's
+/// relevant `Document`'s origin is not same origin-domain with the entry settings object's origin.
+/// See: https://html.spec.whatwg.org/#dom-location-href
+///
```

Comment:
Hmm... I wonder if it would be possible to write a test (which would run locally, totally offline and without setting up dummy servers to fetch something cross-origin) to check that one of the methods actually throws a security error?

These are simple enough that I'm fine with not having a test, but *if* you could think of a way to to write such a test I'd certainly be very nice. (:

Reply:
I have absolutely no clue... haha.

It would be nice to be able to spin up such tests easily without something like selenium. With selenium it would probably be possible though...